redbiom/metadata.py

```diff
@@ -0,0 +1,165 @@
+                               index_label=index_column)
+
+        c = conn.cursor()
+
+        # In general we wouldn't want to format our query in this way because
+        # it leaves us open to sql injection, but it seems acceptable here for
+        # a few reasons:
+        # 1) This is a throw-away database which we're just creating to have
+        #    access to the query language, so any malicious behavior wouldn't
+        #    impact any data that isn't temporary
+        # 2) The substitution syntax recommended in the docs doesn't allow
+        #    us to specify complex `where` statements, which is what we need to
+        #    do here. For example, we need to specify things like:
+        #        WHERE Subject='subject-1' AND SampleType='gut'
+        #    but their qmark/named-style syntaxes only supports substition of
+        #    variables, such as:
+        #        WHERE Subject=?
+        # 3) sqlite3.Cursor.execute will only execute a single statement so
+        #    inserting multiple statements
+        #    (e.g., "Subject='subject-1'; DROP...") will result in an
+        #    OperationalError being raised.
+        query = ('SELECT "{0}" FROM metadata WHERE {1} GROUP BY "{0}" '
+                 'ORDER BY "{0}";'.format(index_column, where))
+
+        try:
+            c.execute(query)
+        except sqlite3.OperationalError:
+            conn.close()
+            raise ValueError("Selection of IDs failed with query:\n %s"
+                             % query)
+
+        ids = set(c.fetchall())
+        conn.close()
+        return ids
+
+
+def samples(sample_values, criteria):
+    """Select samples based on specified criteria
+
+    Parameters
+    ----------
+    sample_values : pandas.Series
+        A series indexed by the Sample ID and valued by something.
+    criteria : str
+        Selection criteria. Simple logic can be specified, but cannot be
+        chained. The following operators are available:
+
+            {<, >, in, notin}
+
+        For example, to keep samples with a value less than 5, the following
+        form works: "< 5". To keep samples matching a discrete set of possible
+        states, use the "in" operator and denote the valid states with a comma.
+        Quotes are possible as well, for instance, "in foo,'bar baz" will keep
+        samples whose value are either "foo" or "bar baz".
+
+        If no operator is specified, it is assumed an exact string match of the
+        value is to be performed.
+
+    Returns
+    -------
+    generator of str
+        Yields the sample IDs which meet the criteria.
+
+    Raises
+    ------
+    ValueError
+        If the criteria cannot be parsed.
+        If the > or < operator is used, and the right hand side of the
+            criteria do not appear to be numeric.
+    """
+    import shlex
+    from redbiom.util import float_or_nan
+    tokens = list(shlex.shlex(criteria))
+    if len(tokens) > 1:
```

Comment:
why do we need the special case for == 1? shouldn't `if operator is None:` cover that case?